tests/test_libe57.py

```diff
@@ -0,0 +1,193 @@
+import pytest
+import numpy
+from pye57 import libe57, E57
+from pye57.e57 import SUPPORTED_POINT_FIELDS
+from pye57.utils import get_node
+
+from .test_main import sample_data
+
+BUFFER_TYPES = {
+    libe57.FloatNode: 'd',
+    libe57.IntegerNode: 'l',
+    libe57.ScaledIntegerNode: 'd'
+}
+
+def make_buffer(node_field, capacity):
+    node_field_type = type(node_field)
+    if node_field_type not in BUFFER_TYPES:
+        raise ValueError("Unsupported field type!")
+    
+    buffer_type = BUFFER_TYPES[node_field_type]
+
+    np_array = numpy.empty(capacity, buffer_type)
+    buffer = libe57.SourceDestBuffer(
+        node_field.destImageFile(), node_field.elementName(), np_array, capacity, True, True)
+    return np_array, buffer
+
+def make_buffers(node: libe57.StructureNode, capacity):
+    data = {}
+    buffers = libe57.VectorSourceDestBuffer()
+    for i in range(0, node.childCount()):
+        field = get_node(node, i)
+        d, b = make_buffer(field, capacity)
+        data[field.elementName()] = d
+        buffers.append(b)
+    return data, buffers
+
+def copy_compressed_vector_data(in_node, out_node):
+    chunk_size = 100000
+    
+    in_prototype = libe57.StructureNode(in_node.prototype())
+    out_prototype = libe57.StructureNode(out_node.prototype())
+    
+    in_data, in_buffers = make_buffers(in_prototype, chunk_size)
+    out_data, out_buffers = make_buffers(out_prototype, chunk_size)
+
+    in_reader = in_node.reader(in_buffers)
+    out_writer = out_node.writer(out_buffers)
+
+    n_points = in_node.childCount()
+    current_index = 0
+    while current_index != n_points:
+        current_chunk = min(n_points - current_index, chunk_size)
+
+        in_reader.read()
+        for field in in_data:
+            out_data[field][:current_chunk] = in_data[field][:current_chunk]
+
+        out_writer.write(current_chunk)
+
+        current_index += current_chunk
+    
+    in_reader.close()
+    out_writer.close()
+
+def copy_blob_data(in_node, out_node):
+    chunk_size = 100000
+
+    byte_count = in_node.byteCount()
+    blob_buffer = numpy.empty(chunk_size, numpy.ubyte)
+    current_index = 0
+    while current_index != byte_count:
+        current_chunk = min(byte_count - current_index, chunk_size)
+
+        in_node.read(blob_buffer, current_index, current_chunk)
+        out_node.write(blob_buffer, current_index, current_chunk)
+
+        current_index += current_chunk
+
+def copy_node(node, dest_image):
```

Comment:
question: would it make sense to add this function to the main pye57 library? For instance in the utils namespace ?